lib/BackgroundJob/EmailNotification.php

```
@@ -123,22 +123,33 @@ protected function runStep($limit, $sendTime) {
 		// Send Email
 		$default_lang = $this->config->getSystemValue('default_language', 'en');
 		$defaultTimeZone = date_default_timezone_get();
+
+		$sentMailForUsers = [];
+
 		foreach ($affectedUsers as $user) {
 			$uid = $user['uid'];
 			if (empty($user['email'])) {
 				// The user did not setup an email address
 				// So we will not send an email but still discard the queue entries
 				$this->logger->debug("Couldn't send notification email to user '$uid' (email address isn't set for that user)", ['app' => 'activity']);
+				$sentMailForUsers[] = $uid;
 				continue;
 			}
 
 			$language = (!empty($userLanguages[$uid])) ? $userLanguages[$uid] : $default_lang;
 			$timezone = (!empty($userTimezones[$uid])) ? $userTimezones[$uid] : $defaultTimeZone;
-			$this->mqHandler->sendEmailToUser($uid, $user['email'], $language, $timezone, $sendTime);
+
+			try {
+				$this->mqHandler->sendEmailToUser($uid, $user['email'], $language, $timezone, $sendTime);
+				$sentMailForUsers[] = $uid;
+			} catch (\Exception $e) {
```

isn't that too tight an exception ? we likely only want to catch and skip validation exceptions.

If the mail server (SMTP) isn't available, we should directly abort instead of continuing the queue.

This was a bit the idea in my PR, except that instead of inventing a new exception in the mailer I just did the validation directly in activity. Probably not the best idea. Should we invent a new exception in our mailer code then ?